temporalcli/commands.operator_namespace.go

```diff
@@ -0,0 +1,335 @@
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create namespace: %w", err)
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s successfully registered.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDeleteCommand) run(cctx *CommandContext, args []string) error {
+	nsName := args[0]
+	yes, err := cctx.promptString(
+		color.RedString("Are you sure you want to delete namespace %s? Type namespace name to confirm:", nsName),
+		nsName,
+		c.Yes)
+	if err != nil {
+		return err
+	}
+
+	if !yes {
+		return fmt.Errorf("user denied confirmation")
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.OperatorService().DeleteNamespace(cctx, &operatorservice.DeleteNamespaceRequest{
+		Namespace: nsName,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to delete namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	cctx.Printer.Println(color.GreenString("Namespace %s has been deleted.", nsName))
+	return nil
+}
+
+func (c *TemporalOperatorNamespaceDescribeCommand) run(cctx *CommandContext, args []string) error {
+	nsName, nsID, err := getNamespaceFromIDArgs(cctx, c.NamespaceId, args)
+	if err != nil {
+		return err
+	}
+
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.WorkflowService().DescribeNamespace(cctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: nsName,
+		Id:        nsID,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to describe namespace: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+	return printNamespaces(cctx, resp)
+}
+
+func (c *TemporalOperatorNamespaceListCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+
+	var nextPageToken []byte
+	for {
+		resp, err := cl.WorkflowService().ListNamespaces(cctx, &workflowservice.ListNamespacesRequest{
+			NextPageToken: nextPageToken,
+			PageSize:      100,
+		})
+		if err != nil {
+			return fmt.Errorf("failed listing namespaces: %w", err)
+		}
+
+		if cctx.JSONOutput {
+			// For JSON we are going to dump one line of JSON per execution
+			_ = cctx.Printer.PrintStructured(resp.Namespaces, printer.StructuredOptions{})
+		} else {
+			_ = printNamespaces(cctx, resp.Namespaces...)
+		}
+
+		nextPageToken = resp.GetNextPageToken()
+		if len(nextPageToken) == 0 {
+			return nil
+		}
+	}
+}
+
+func (c *TemporalOperatorNamespaceUpdateCommand) run(cctx *CommandContext, args []string) error {
+	nsName := args[0]
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	var updateRequest *workflowservice.UpdateNamespaceRequest
+
+	if c.PromoteGlobal {
```

Comment:
If these are mutually exclusive options, we should probably error if multiple are provided